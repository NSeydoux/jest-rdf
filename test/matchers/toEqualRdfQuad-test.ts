import {blankNode, namedNode, quad} from "@rdfjs/data-model";
import * as RDF from "rdf-js";
import "../../index";

describe('#toEqualRdfQuad', () => {
  it('should succeed for equal quads', () => {
    return expect(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g'),
    )).toEqualRdfQuad(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g'),
    ));
  });

  it('should succeed for equal generalized quads', () => {
    return expect(quad<RDF.BaseQuad>(
      blankNode('s'),
      blankNode('p'),
      blankNode('o'),
      blankNode('g'),
    )).toEqualRdfQuad(quad<RDF.BaseQuad>(
      blankNode('s'),
      blankNode('p'),
      blankNode('o'),
      blankNode('g'),
    ));
  });

  it('should not fail for equal quads', () => {
    return expect(() => expect(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g'),
    )).not.toEqualRdfQuad(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g'),
    ))).toThrowErrorMatchingSnapshot();
  });

  it('should not succeed for quads with different subjects', () => {
    return expect(quad(
      namedNode('s1'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g'),
    )).not.toEqualRdfQuad(quad(
      namedNode('s2'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g'),
    ));
  });

  it('should not succeed for quads with different predicates', () => {
    return expect(quad(
      namedNode('s'),
      namedNode('p1'),
      namedNode('o'),
      namedNode('g'),
    )).not.toEqualRdfQuad(quad(
      namedNode('s'),
      namedNode('p2'),
      namedNode('o'),
      namedNode('g'),
    ));
  });

  it('should not succeed for quads with different objects', () => {
    return expect(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o1'),
      namedNode('g'),
    )).not.toEqualRdfQuad(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o2'),
      namedNode('g'),
    ));
  });

  it('should not succeed for quads with different graphs', () => {
    return expect(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g1'),
    )).not.toEqualRdfQuad(quad(
      namedNode('s'),
      namedNode('p'),
      namedNode('o'),
      namedNode('g2'),
    ));
  });
});
