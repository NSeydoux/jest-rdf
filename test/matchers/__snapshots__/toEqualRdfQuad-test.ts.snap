// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toEqualRdfQuad should not fail for equal quads 1`] = `
"expected
  {\\"subject\\":\\"s\\",\\"predicate\\":\\"p\\",\\"object\\":\\"o\\",\\"graph\\":\\"g\\"}
not to equal
  {\\"subject\\":\\"s\\",\\"predicate\\":\\"p\\",\\"object\\":\\"o\\",\\"graph\\":\\"g\\"}"
`;
